RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -167,20 +167,36 @@ math::XYZPoint HGCalImagingAlgo::calculatePosition(std::vector<KDNode> &v){
   float y = 0.;
   float z = 0.;
   unsigned int v_size = v.size();
+  bool haloOnlyCluster = true;
 
+  // check if haloOnlyCluster
   for (unsigned int i = 0; i < v_size; i++){
     if(!v[i].data.isHalo){
-      total_weight += v[i].data.weight;
-      x += v[i].data.x*v[i].data.weight;
-      y += v[i].data.y*v[i].data.weight;
-      z += v[i].data.z*v[i].data.weight;
+      haloOnlyCluster = false;
```

Comment:
You've turned one loop into three loops... at minimum you can `break` this first loop at this point, but I would suggest combining all three loops. Just build up the total weight and find the index of the max element at the same time, then use whichever one you need based on the flag.